okhttp-logging-interceptor/src/main/java/com/squareup/okhttp/logging/HttpLoggingInterceptor.java

```diff
@@ -179,7 +179,7 @@ public Level getLevel() {
       }
 
       String endMessage = "--> END " + request.method();
-      if (logBody && hasRequestBody) {
+      if (logBody && hasRequestBody && logBody(request.headers())) {
```

Comment:
This would be especially nice since we use the buffered body to get its size, which is otherwise unavailable due to lack of a header.